src/codeu/chat/server/Server.java

```diff
@@ -276,11 +305,80 @@ public void run() {
         final ConversationHeader conversation = view.findConversation(conversationId);
         final Message message = view.findMessage(messageId);
         relay.write(id,
-                    secret,
-                    relay.pack(user.id, user.name, user.creation),
-                    relay.pack(conversation.id, conversation.title, conversation.creation),
-                    relay.pack(message.id, message.content, message.creation));
+                secret,
+                relay.pack(user.id, user.name, user.creation),
+                relay.pack(conversation.id, conversation.title, conversation.creation),
+                relay.pack(message.id, message.content, message.creation));
       }
     };
   }
+
+  private void process(String line) throws IOException {
```

Comment:
this logic should be in a static Util function , like TransactionLogUtil.readLogs() ...